doc/source/whatsnew/v1.0.0.rst

```diff
@@ -39,8 +39,59 @@ Backwards incompatible API changes
 
 .. _whatsnew_1000.api.other:
 
-- :class:`pandas.core.groupby.GroupBy.transform` now raises on invalid operation names (:issue:`27489`).
--
+Groupby.transform(name) accepts only aggregation names
```

Comment:
At a glance, this makes it sound like `.transform(callable)` isn't allowed anymore. Perhaps we can say GroupBy.transform now validates aggregation names.

Reply:
The body explicitly states the callable behavior isn't changed. Changed the title to avoid confusion.